<p>So, after a bit of investigation it turns out the exceptions that were being thrown as a <code>SOAPFaultException</code> were not coming from where I thought they were. They actually were not thrown as <code>MyCustomException</code>. So, to solve this for all cases I took advantage of Spring AOP and created an Around aspect to wrap all of my calls to the web service implementation layer.</p>

<p>In this aspect I catch any exception thrown and wrap it in a <code>MyCustomException</code> before rethrowing it. My client can then handle the exception properly.</p>
